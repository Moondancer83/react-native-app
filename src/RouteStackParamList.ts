export type RootStackParamList = {
  Splash: undefined;
  Counter: undefined;
  Results: undefined;
};