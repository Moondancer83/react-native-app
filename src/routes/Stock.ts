export interface Stock {
  symbol: string;
  name: string;
  priceCurrent: number;
  priceOpening: number;
  priceHeight: number;
  priceLow: number;
  history: {[date: string]: number};
}
