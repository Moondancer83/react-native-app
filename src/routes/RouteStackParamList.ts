import {Stock} from '../service/Stock';

export type RootStackParamList = {
  Splash: undefined;
  Search: undefined;
  Detail: {stock: Stock};
};
