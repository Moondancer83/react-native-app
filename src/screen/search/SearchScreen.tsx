import React, {useState} from 'react';
import {Text, TextInput, TouchableOpacity, View} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';

import {Stock} from '../../service/Stock';
import {RootStackParamList} from '../../routes/RouteStackParamList';
import {getSymbol, searchStock} from '../../service/stockFacade';
import ScreenFrame from '../ScreenFrame';
import {style} from './SearchScreen.style';
import {baseStyles, colors} from '../BaseStyles';
import {RouteProp} from '@react-navigation/native';
import {SymbolDTO} from '../../service/finnhub/FinnHubApi';

interface Props {
  navigation: StackNavigationProp<RootStackParamList, 'Search'>;
  route: RouteProp<RootStackParamList, 'Search'>;
}

export default function SearchScreen(props: Props) {
  const [symbol, setSymbol] = useState<string>();
  const [error, setError] = useState<boolean>(false);
  const [focus, setFocus] = useState<boolean>(false);

  function getProfile(): SymbolDTO | undefined {
    const {symbols} = props.route.params;
    const profile = getSymbol(symbols, symbol);
    if (!profile) {
      setError(true);
    }
    return profile;
  }

  async function search(profile: SymbolDTO) {
    const result = await searchStock(profile);
    if (result.error || result.stock == null) {
      setError(true);
    } else {
      setError(false);
      const stock: Stock = result.stock;
      props.navigation.navigate('Detail', {stock});
    }
  }

  async function submit() {
    const profile = getProfile();
    if (!profile) {
      return;
    }
    search(profile);
  }

  return (
    <ScreenFrame>
      <View style={error ? style.errorShow : style.error}>
        <Text style={style.errorText}>
          {error ? 'This is not a valid Ticker Symbol' : null}
        </Text>
      </View>
      <View style={style.inputSection}>
        <Text style={baseStyles.title2}>Enter Ticker Symbol</Text>
        <TextInput
          autoCapitalize={'characters'}
          editable={true}
          onChangeText={(text) => {
            setSymbol(text);
            setError(false);
          }}
          onFocus={() => setFocus(true)}
          onBlur={() => setFocus(false)}
          style={
            error ? style.inputError : focus ? style.inputFocused : style.input
          }
          placeholder={'VOO'}
          placeholderTextColor={colors.lightGray}
          onSubmitEditing={submit}
        />
      </View>
      <View>
        <TouchableOpacity
          disabled={!symbol}
          onPress={submit}
          style={symbol ? style.button : style.buttonDisabled}>
          <Text style={style.buttonText}>GO</Text>
        </TouchableOpacity>
      </View>
    </ScreenFrame>
  );
}
