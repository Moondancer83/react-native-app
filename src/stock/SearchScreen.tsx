import React, {useState} from 'react';
import {Button, Text, TextInput, View} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';

import {Stock} from '../routes/Stock';
import {RootStackParamList} from '../routes/RouteStackParamList';
import {searchStock} from './service/stockFacade';
import ScreenFrame from '../core/ScreenFrame';
import {style} from './Screen.style';

interface Props {
  navigation: StackNavigationProp<RootStackParamList, 'Search'>;
}

export default function SearchScreen(props: Props) {
  const [symbol, setSymbol] = useState<string>();
  const [error, setError] = useState<boolean>(false);

  return (
    <ScreenFrame>
      <View style={style.view}>
        {error ? (
          <Text style={{backgroundColor: 'red', color: 'white'}}>
            This is not a valid Ticker Symbol
          </Text>
        ) : null}
        <Text>Enter Ticker Symbol</Text>
        <TextInput
          editable={true}
          onChangeText={(text) => {
            setSymbol(text);
            setError(false);
          }}
          style={{
            width: '80%',
            height: 40,
            borderColor: error ? 'red' : 'gray',
            borderWidth: 1,
            textAlign: 'center',
          }}
        />
        <Button
          title={'GO'}
          onPress={async () => {
            const result = await searchStock(symbol);
            if (result.error || result.stock == null) {
              setError(true);
            } else {
              setError(false);
              const stock: Stock = result.stock;
              props.navigation.navigate('Detail', {stock});
            }
          }}
        />
      </View>
    </ScreenFrame>
  );
}
