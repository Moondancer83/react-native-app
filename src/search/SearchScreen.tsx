import React, {useState} from 'react';
import {Text, TextInput, TouchableNativeFeedback, View} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';

import {Stock} from '../routes/Stock';
import {RootStackParamList} from '../routes/RouteStackParamList';
import {searchStock} from '../stock/service/stockFacade';
import ScreenFrame from '../screen/ScreenFrame';
import {style} from './SearchScreen.style';
import {baseStyles} from '../screen/BaseStyles';

interface Props {
  navigation: StackNavigationProp<RootStackParamList, 'Search'>;
}

export default function SearchScreen(props: Props) {
  const [symbol, setSymbol] = useState<string>();
  const [error, setError] = useState<boolean>(false);
  const [inputStyle, setInputStyle] = useState(style.input);
  const [errorStyle, setErrorStyle] = useState(style.error);
  const [buttonStyle, setButtonStyle] = useState(style.buttonDisabled);

  return (
    <ScreenFrame>
      <View style={errorStyle}>
        <Text style={style.errorText}>
          {error ? 'This is not a valid Ticker Symbol' : null}
        </Text>
      </View>
      <View style={style.inputSection}>
        <Text style={baseStyles.title2}>Enter Ticker Symbol</Text>
        <TextInput
          editable={true}
          onChangeText={(text) => {
            setSymbol(text);
            setErrorStyle(style.error);
            setError(false);
            setInputStyle(style.inputFocused);
            if (text) {
              setButtonStyle(style.button);
            } else {
              setButtonStyle(style.buttonDisabled);
            }
          }}
          onFocus={() => setInputStyle(style.inputFocused)}
          style={inputStyle}
        />
      </View>
      <View style={buttonStyle}>
        <TouchableNativeFeedback disabled={!symbol}>
          <Text
            style={style.buttonText}
            onPress={async () => {
              const result = await searchStock(symbol);
              if (result.error || result.stock == null) {
                setError(true);
                setErrorStyle(style.errorShow);
                setInputStyle(style.inputError);
              } else {
                setError(false);
                setInputStyle(style.inputFocused);
                const stock: Stock = result.stock;
                props.navigation.navigate('Detail', {stock});
              }
            }}>
            GO
          </Text>
        </TouchableNativeFeedback>
      </View>
    </ScreenFrame>
  );
}
