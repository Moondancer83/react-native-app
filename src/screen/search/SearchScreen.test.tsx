import React from 'react';
import renderer, {act} from 'react-test-renderer';

import SearchScreen from './SearchScreen';
import {StackNavigationProp} from '@react-navigation/stack';
import {RootStackParamList} from '../../routes/RouteStackParamList';
import {RouteProp} from '@react-navigation/native';
import {style} from './SearchScreen.style';

test('Test SearchScreen input style', () => {
  const navigation:
    | StackNavigationProp<RootStackParamList, 'Search'>
    | any = {};
  const route: RouteProp<RootStackParamList, 'Search'> | any = {};

  const component = renderer.create(
    <SearchScreen navigation={navigation} route={route} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(
    tree.children[1].children[0].children[1].props.style.borderColor,
  ).toEqual('#B2B2B2');

  tree.children[1].children[0].children[1].props.onFocus();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(
    tree.children[1].children[0].children[1].props.style.borderColor,
  ).toEqual('#000000');

  tree.children[1].children[0].children[1].props.onBlur();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(
    tree.children[1].children[0].children[1].props.style.borderColor,
  ).toEqual('#B2B2B2');
});

test('Test SearchScreen button is disabled when input is empty', () => {
  const navigation:
    | StackNavigationProp<RootStackParamList, 'Search'>
    | any = {};
  const route: RouteProp<RootStackParamList, 'Search'> | any = {};

  const component = renderer.create(
    <SearchScreen navigation={navigation} route={route} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[2].children[0].props.style.backgroundColor).toEqual(
    style.buttonDisabled.backgroundColor,
  );

  act(() => {
    tree.children[1].children[0].children[1].props.onChangeText('AAPL');
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[2].children[0].props.style.backgroundColor).toEqual(
    style.button.backgroundColor,
  );

  act(() => {
    tree.children[1].children[0].children[1].props.onChangeText('');
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[2].children[0].props.style.backgroundColor).toEqual(
    style.buttonDisabled.backgroundColor,
  );
});
