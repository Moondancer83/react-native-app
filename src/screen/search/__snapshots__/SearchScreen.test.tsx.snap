// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchScreen button is disabled when input is empty 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 71,
        "paddingBottom": 28,
        "paddingTop": 28,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next",
          "fontSize": 15,
          "fontWeight": "500",
          "textAlign": "center",
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Enter Ticker Symbol
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="VOO"
        placeholderTextColor="#E5E5E5"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#B2B2B2",
            "borderRadius": 3,
            "borderWidth": 1,
            "fontFamily": "Avenir Next",
            "fontSize": 28,
            "fontWeight": "600",
            "height": 108,
            "lineHeight": 40,
            "marginBottom": 28,
            "marginTop": 28,
            "paddingBottom": 28,
            "paddingTop": 28,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E5E5E5",
          "opacity": 1,
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        GO
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test SearchScreen button is disabled when input is empty 2`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 71,
        "paddingBottom": 28,
        "paddingTop": 28,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next",
          "fontSize": 15,
          "fontWeight": "500",
          "textAlign": "center",
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Enter Ticker Symbol
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="VOO"
        placeholderTextColor="#E5E5E5"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#B2B2B2",
            "borderRadius": 3,
            "borderWidth": 1,
            "fontFamily": "Avenir Next",
            "fontSize": 28,
            "fontWeight": "600",
            "height": 108,
            "lineHeight": 40,
            "marginBottom": 28,
            "marginTop": 28,
            "paddingBottom": 28,
            "paddingTop": 28,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#000000",
          "opacity": 1,
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        GO
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test SearchScreen button is disabled when input is empty 3`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 71,
        "paddingBottom": 28,
        "paddingTop": 28,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next",
          "fontSize": 15,
          "fontWeight": "500",
          "textAlign": "center",
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Enter Ticker Symbol
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="VOO"
        placeholderTextColor="#E5E5E5"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#B2B2B2",
            "borderRadius": 3,
            "borderWidth": 1,
            "fontFamily": "Avenir Next",
            "fontSize": 28,
            "fontWeight": "600",
            "height": 108,
            "lineHeight": 40,
            "marginBottom": 28,
            "marginTop": 28,
            "paddingBottom": 28,
            "paddingTop": 28,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E5E5E5",
          "opacity": 1,
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        GO
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test SearchScreen input style 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 71,
        "paddingBottom": 28,
        "paddingTop": 28,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next",
          "fontSize": 15,
          "fontWeight": "500",
          "textAlign": "center",
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Enter Ticker Symbol
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="VOO"
        placeholderTextColor="#E5E5E5"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#B2B2B2",
            "borderRadius": 3,
            "borderWidth": 1,
            "fontFamily": "Avenir Next",
            "fontSize": 28,
            "fontWeight": "600",
            "height": 108,
            "lineHeight": 40,
            "marginBottom": 28,
            "marginTop": 28,
            "paddingBottom": 28,
            "paddingTop": 28,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E5E5E5",
          "opacity": 1,
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        GO
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test SearchScreen input style 2`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 71,
        "paddingBottom": 28,
        "paddingTop": 28,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next",
          "fontSize": 15,
          "fontWeight": "500",
          "textAlign": "center",
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Enter Ticker Symbol
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="VOO"
        placeholderTextColor="#E5E5E5"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#000000",
            "borderRadius": 3,
            "borderWidth": 1,
            "fontFamily": "Avenir Next",
            "fontSize": 28,
            "fontWeight": "600",
            "height": 108,
            "lineHeight": 40,
            "marginBottom": 28,
            "marginTop": 28,
            "paddingBottom": 28,
            "paddingTop": 28,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E5E5E5",
          "opacity": 1,
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        GO
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test SearchScreen input style 3`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 71,
        "paddingBottom": 28,
        "paddingTop": 28,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next",
          "fontSize": 15,
          "fontWeight": "500",
          "textAlign": "center",
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Enter Ticker Symbol
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="VOO"
        placeholderTextColor="#E5E5E5"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#B2B2B2",
            "borderRadius": 3,
            "borderWidth": 1,
            "fontFamily": "Avenir Next",
            "fontSize": 28,
            "fontWeight": "600",
            "height": 108,
            "lineHeight": 40,
            "marginBottom": 28,
            "marginTop": 28,
            "paddingBottom": 28,
            "paddingTop": 28,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E5E5E5",
          "opacity": 1,
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir Next",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        GO
      </Text>
    </View>
  </View>
</View>
`;
